### PR TITLE
Added Discord prompt param

### PR DESCRIPTION
**Pull request by nerrixDE**
As documented [here](https://discord.com/developers/docs/topics/oauth2#authorization-code-grant-authorization-url-example) this would be good to get implemented.

### CODE REVIEW

**Comment by daenney** [comment]
I don't think this should default to `consent`. If the user hasn't authorized previously or the scopes have changed, even with a prompt of `None` Discord will still prompt. A default of `consent` will always cause Discord to ask even without a scope change, which is not the experience most people want or expect out of the box.

I'd probably change the implementation to do what we do in the Google provider, with a `reprompt_consent` that defaults to `False` but can be set to `True` and then an `if reprompt_consent` add it to the `authorization_url`.

**Comment by nerrixDE** [comment]
@daenney I agree, however Discord's default is `consent` and right now it is set to `consent` implicitly. Thus setting it to `None` on default would change existing behaviour (There is only `None` and `consent`)

**Comment by daenney** [comment]
Ah I see. It wasn't clear to me from the docs that if it's omitted they default it to consent instead of none.

I'd still like to see this changed to take a boolean of  `reprompt_consent` instead, to keep it inline with the 2 other providers where we do this (Google and Zoho), but in that case default it to `True` instead to keep the current behaviour intact.

**Comment by nerrixDE** [comment]
@daenney Good point however I would personally prefer to keep the field as it is defined by the API (Azure provider uses `prompt` as well). Also putting a boolean wrapper around it may cause issues in case Discord decides to add additional prompt-modes in future.

**Comment by daenney** [comment]
Fair point, I had missed the Azure one.

**Comment by daenney** [comment]
Can you add an additional test though? Right now only the behaviour of explicitly setting to to `None` is tested, which is no longer the default of the blueprint.

File: tests/contrib/test_discord.py [hunk 1]
```diff
@@ -39,6 +39,24 @@ def test_blueprint_factory():
     assert discord_bp.token_url == "https://discord.com/api/oauth2/token"
```
Comment (daenney) [inline tests/contrib/test_discord.py:40]: You should add an `assert` to `test_blueprint_factory()` too, to ensure the default prompt value of consent is now there.

File: tests/contrib/test_discord.py [hunk 2]
```diff
@@ -66,6 +84,7 @@ def test_context_local(make_app):
         "bar2",
         redirect_to="url2",
         storage=MemoryStorage({"access_token": "app2"}),
+        prompt=None,
```
Comment (daenney) [inline tests/contrib/test_discord.py:87]: I don't think this one is necessary anymore?

**Review by daenney — APPROVED** [review]
